lib/cartopy/mpl/geoaxes.py

```diff
@@ -532,21 +528,22 @@ def draw(self, renderer=None, **kwargs):
 
     def _update_title_position(self, renderer):
         super()._update_title_position(renderer)
-        if not self._gridliners:
+
+        from cartopy.mpl.gridliner import Gridliner
+        gridliners = [a for a in self.artists if isinstance(a, Gridliner)]
+        if not gridliners:
             return
 
         if self._autotitlepos is not None and not self._autotitlepos:
             return
```

Comment:
Minor, but possibly put this check first to short-circuit out of it before iterating through artists?